client/webserver/webserver.go

```diff
@@ -284,7 +284,11 @@ func New(cfg *Config) (*WebServer, error) {
 
 	// Middleware
 	if log.Level() == dex.LevelTrace {
-		mux.Use(middleware.Logger)
+		mux.Use(
+			middleware.RequestLogger(
+				&middleware.DefaultLogFormatter{Logger: &chiLogger{logger: log}},
```

Comment:
With `middleware.Logger` on Windows, `NoColor` was false.

However, now that this is going to a log file rather than a color terminal, we're wrecking the log file on all platforms:

```
2023-03-20 09:46:28.274 [INF] WEB: Web server listening on 127.0.0.3:5758
2023-03-20 09:46:30.012 [TRC] WEB: ESC[36m"ESC[0mESC[35;1mGET ESC[0mESC[36mhttp://127.0.0.3:5758/ HTTP/1.1" ESC[0mfrom 127.0.0.1:40532 - ESC[36;1m303ESC[0mESC[34;1m 36BESC[0m in ESC[32m73.535µsESC[0m
2023-03-20 09:46:30.045 [TRC] WEB: ESC[36m"ESC[0mESC[35;1mGET ESC[0mESC[36mhttp://127.0.0.3:5758/register HTTP/1.1" ESC[0mfrom 127.0.0.1:40532 - ESC[32;1m200ESC[0mESC[34;1m 22565BESC[0m in ESC[32m769.354µsESC[0m
2023-03-20 09:46:30.079 [TRC] WEB: ESC[36m"ESC[0mESC[35;1mGET ESC[0mESC[36mhttp://127.0.0.3:5758/css/style.css?v=YYNQOy0 HTTP/1.1" ESC[0mfrom 127.0.0.1:40532 - ESC[32;1m200ESC[0mESC[34;1m 167916BESC[0m in ESC[32m212.038µsESC[0m
2023-03-20 09:46:30.082 [TRC] WEB: ESC[36m"ESC[0mESC[35;1mGET ESC[0mESC[36mhttp://127.0.0.3:5758/img/coins/z.png HTTP/1.1" ESC[0mfrom 127.0.0.1:40542 - ESC[32;1m200ESC[0mESC[34;1m 2037BESC[0m in ESC[32m2.644381msESC[0m
2023-03-20 09:46:30.180 [TRC] WEB: ESC[36m"ESC[0mESC[35;1mGET ESC[0mESC[36mhttp://127.0.0.3:5758/img/coins/dcr.png HTTP/1.1" ESC[0mfrom 127.0.0.1:40532 - ESC[32;1m200ESC[0mESC[34;1m 2909BESC[0m in ESC[32m73.981µsESC[0m
2023-03-20 09:46:30.181 [TRC] WEB: ESC[36m"ESC[0mESC[35;1mGET ESC[0mESC[36mhttp://127.0.0.3:5758/js/entry.js?v=YYNQOy0 HTTP/1.1" ESC[0mfrom 127.0.0.1:40542 - ESC[32;1m200ESC[0mESC[34;1m 334777BESC[0m in ESC[32m232.084µsESC[0m
2023-03-20 09:46:30.188 [TRC] WEB: ESC[36m"ESC[0mESC[35;1mGET ESC[0mESC[36mhttp://127.0.0.3:5758/img/softened-icon-dark.png HTTP/1.1" ESC[0mfrom 127.0.0.1:40542 - ESC[32;1m200ESC[0mESC[34;1m 3207BESC[0m in ESC[32m77.019µsESC[0m
2023-03-20 09:46:30.206 [TRC] WEB: ESC[36m"ESC[0mESC[35;1mGET ESC[0mESC[36mhttp://127.0.0.3:5758/font/icomoon.ttf?u8i98qq HTTP/1.1" ESC[0mfrom 127.0.0.1:40542 - ESC[32;1m200ESC[0mESC[34;1m 7284BESC[0m in ESC[32m117.21µsESC[0m
2023-03-20 09:46:30.207 [TRC] WEB: ESC[36m"ESC[0mESC[35;1mGET ESC[0mESC[36mhttp://127.0.0.3:5758/font/source-sans-pro-v9-latin-regular.woff HTTP/1.1" ESC[0mfrom 127.0.0.1:40532 - ESC[32;1m200ESC[0mESC[34;1m 17500BESC[0m in ESC[32m88.463µsESC[0m
2023-03-20 09:46:30.246 [TRC] WEB: ESC[36m"ESC[0mESC[35;1mGET ESC[0mESC[36mhttp://127.0.0.3:5758/api/user HTTP/1.1" ESC[0mfrom 127.0.0.1:40532 - ESC[33;1m401ESC[0mESC[34;1m 29BESC[0m in ESC[32m36.82µsESC[0m
2023-03-20 09:46:30.250 [TRC] WEB: ESC[36m"ESC[0mESC[35;1mGET ESC[0mESC[36mhttp://127.0.0.3:5758/img/favicon.png?v=AK4XS4 HTTP/1.1" ESC[0mfrom 127.0.0.1:40532 - ESC[32;1m200ESC[0mESC[34;1m 2550BESC[0m in ESC[32m93.696µsESC[0m
2023-03-20 09:46:30.264 [TRC] WEB: ESC[36m"ESC[0mESC[35;1mGET ESC[0mESC[36mhttp://127.0.0.3:5758/ws HTTP/1.1" ESC[0mfrom 127.0.0.1:40544 - ESC[34;1m000ESC[0mESC[34;1m 0BESC[0m in ESC[32m107.031µsESC[0m
2023-03-20 09:46:30.264 [DBG] WEB[WS]: New websocket client 127.0.0.1:40544
2023-03-20 09:46:30.264 [TRC] WEB[WS][127.0.0.1:40544]: Starting websocket messaging with peer 127.0.0.1:40544
```

That's terrible both visually and for grepping.  So if we go ahead with this PR's change, we need to set `NoColor: true` always.

1. That kinda sucks
2. is it very valuable to put every request for a png file in the log file?

Reply:
> is it very valuable to put every request for a png file in the log file?

Probably not, but I'm not sure that not writing HTTP requests to logs at all is better. Anyway, chi output to stdout due to that coloring issue.